packages/tpetra/core/src/Tpetra_Details_residual.hpp

```diff
@@ -211,9 +211,12 @@ void localResidual(const CrsMatrix<SC,LO,GO,NO> &  A,
   int team_size = -1;
   int vector_length = -1;
   int64_t rows_per_thread = -1;
+  
+  int64_t numLocalRows = A_lcl.numRows ();
+  int64_t myNnz = A_lcl.nnz();
 
-  const int64_t rows_per_team = 
-    residual_launch_parameters<execution_space>(A_lcl.numRows (), A_lcl.nnz (), rows_per_thread, team_size, vector_length);
+  int64_t rows_per_team = 
```

Comment:
I have some vague recollection that `const` is related to this compiler bug somehow.

Reply:
the nested lambda issue was fixed by removing const from any captured variables.